Neos.Fusion/Classes/Core/Cache/ParserCacheIdentifierTrait.php

```diff
@@ -26,19 +26,35 @@ private function getCacheIdentifierForDslCode(string $identifier, string $code):
         return 'dsl_' . $identifier . '_' . md5($code);
     }
 
+
     /**
      * creates a comparable hash of the absolute, resolved $fusionFileName
      *
      * @throws \InvalidArgumentException
      */
     private function getCacheIdentifierForFile(string $fusionFileName): string
     {
-        $realPath = realpath($fusionFileName);
-        if ($realPath === false) {
-            throw new \InvalidArgumentException("Couldn't resolve realpath for: '$fusionFileName'");
+        if (self::isRealPath($fusionFileName)) {
+            $realPath = $fusionFileName;
+        } else {
+            $realPath = realpath($fusionFileName);
+            if ($realPath === false) {
+                throw new \InvalidArgumentException("Couldn't resolve realpath for: '$fusionFileName'");
+            }
         }
-
         $realFusionFilePathWithoutRoot = str_replace(FLOW_PATH_ROOT, '', $realPath);
         return 'file_' . md5($realFusionFilePathWithoutRoot);
     }
+
+    private static function isRealPath(string $path): bool
+    {
+        $isAbsoluteWindowsPath = PHP_OS_FAMILY === 'Windows' && preg_match('`^[a-zA-Z]:/[^/]`', $path) === 1;
```

Comment:
IIRC we use other ways to determine the OS (`DIRECTORY_SEPARATOR` ?) and we should not mix them

Reply:
We discussed that this is fine and we will gradually move towards this more readable check.